samza-core/src/main/scala/org/apache/samza/config/RegExTopicGenerator.scala

```diff
@@ -97,19 +95,13 @@ class RegExTopicGenerator extends ConfigRewriter with Logging {
     new MapConfig(((keysAndValsToAdd ++ config.asScala) += inputStreams).asJava)
   }
 
-  def getTopicsFromZK(rewriterName: String, config: Config): Seq[String] = {
-    val systemName = config
+  def getTopicsFromStreamMetadataCache(rewriterName: String, config: Config): Seq[String] = {
```

Comment:
why are we changing this to use StreamMetadatatCache? 
Is this necessary for the fix?

Reply:
Its neccessary because it allows the regex-expansion to be Kafka-agnostic, which allows the regexGenerator to be moved to samza-core (so that the job-model-manager can invoke it). 

Currently, the regexGenerator is in samza-kafka, so the job-model-manager cannot invoke the regexGenerator, without adding a dependency (circular) from samza-core to samza-kafka. 
Also, reading topic info from Zk without using a documented API, bypassing the broker, seems kinda rakish.

Reply:
1. Can we call `getAllSystemStreams` directly on the system admin?
2. SystemAdmin(s) need to started/stopped before/after use.